### PR TITLE
fix(OHIFv3):Fixed a static wado query failing to match modalities in …

### PR DESCRIPTION
**Pull request by wayfarer3130**
Updated the StaticWadoClient to correctly handle modalities in studies lists, patient names and date comparisons.

### CODE REVIEW

File: extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js [hunk 1]
```diff
@@ -36,14 +37,41 @@ export default class StaticWadoClient extends api.DICOMwebClient {
...
+    if (typeof (actual) == 'string') {
+      return actual.indexOf(desired) != -1;
+    }
+    return desired === actual;
+  }
+
+  compareDateRange(range, value) {
+    if (!value) return true;
+    const dash = range.indexOf('-');
+    if (dash == -1) return this.compareValues(range, value);
```
Comment (sedghi) [inline extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js:59]: ===

File: extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js [hunk 2]
```diff
@@ -36,14 +37,41 @@ export default class StaticWadoClient extends api.DICOMwebClient {
     return filtered;
   }
 
+  compareValues(desired, actual) {
```
Comment (sedghi) [inline extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js:40]: can we add some documentation of this recursive function?

File: extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js [hunk 3]
```diff
@@ -36,14 +37,41 @@ export default class StaticWadoClient extends api.DICOMwebClient {
...
     const altKey = StaticWadoClient.filterKeys[key] || key;
     if (!queryParams) return true;
     const testValue = queryParams[key] || queryParams[altKey];
     if (!testValue) return true;
     const valueElem = study[key] || study[altKey];
     if (!valueElem) return false;
+    if (valueElem.vr == 'DA') return this.compareDateRange(testValue, valueElem.Value[0]);
     const value = valueElem.Value;
-    return value === testValue || (value.indexOf && value.indexOf(testValue) >= 0);
+    return this.compareValues(testValue, value) && true;
```
Comment (sedghi) [inline extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.js:75]: what is `&& true` doing? any value `compareValues` return would be the result and don't need to `&& true`

**Comment by sedghi** [comment]
@wayfarer3130 Is this ready for review again?

**Comment by wayfarer3130** [comment]
@sedghi  - yes, this issue is ready to look at again.

**Review by sedghi — APPROVED** [review]

**Pull request merged**